weather/crud.py

```diff
@@ -0,0 +1,108 @@
+from fastapi import HTTPException
+
+from sqlalchemy.orm import Session
+from sqlalchemy.future import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.exc import IntegrityError
+
+from weather import models, schemas
+from weather.utils import exception_handler
+
+
+async def get_all_cities(db: AsyncSession):
+    result = await db.execute(
+        select(models.City)
+    )
+    return result.scalars().all()
```

Comment:
The `get_all_cities` function returns all city records. Ensure that the calling function or endpoint properly handles the list returned by `scalars().all()`.